org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/Member.java

```diff
@@ -394,20 +395,38 @@ public boolean isBinary() {
 }
 protected boolean isMainMethod(IMethod method) throws JavaModelException {
 	if ("main".equals(method.getElementName()) && Signature.SIG_VOID.equals(method.getReturnType())) { //$NON-NLS-1$
-		int flags= method.getFlags();
+		int flags = method.getFlags();
+		Map<String, String> options = method.getJavaProject().getOptions(true);
+		if (JavaCore.ENABLED.equals(options.get(JavaCore.COMPILER_PB_ENABLE_PREVIEW_FEATURES))
```

Comment:
This is not ideal and will need to be touched again when the preview status or version change. Add a new feature to JavaFeature and call its isSupported(options).